smartapps/curb/curb-bridge.src/curb-bridge.groovy

```diff
@@ -0,0 +1,605 @@
+/**
+ *  Curb (Connect)
+ *
+ *  Copyright 2017 Curb
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+include 'asynchttp_v1'
+
+definition(
+    name: "Curb Bridge",
+    namespace: "curb",
+    author: "Curb",
+    description: "App to get usage data from a Curb home energy monitor",
+    category: "",
+    iconUrl: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png",
+    iconX2Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png",
+    iconX3Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png",
+    singleInstance: true
+) {
+    appSetting "clientId"
+}
+
+preferences {
+    page(
+    name: "auth",
+    title: "Curb",
+    nextPage: "",
+    content: "authPage",
+    uninstall: true)
+}
+
+mappings {
+    path("/oauth/initialize") {
+        action: [GET: "oauthInitUrl"]
+    }
+    path("/oauth/callback") {
+        action: [GET: "callback"]
+    }
+}
+
+def installed() {
+    log.debug "Installed with settings: ${settings}"
+}
+
+def updated() {
+    log.debug "Updated with settings: ${settings}"
+    removeChildDevices(getChildDevices())
+    unsubscribe()
+    initialize()
+}
+
+def initialize() {
+    log.debug "Initializing"
+    unschedule()
+    refreshAuthToken()
+    updateSelectedLocationId()
+	  getDevices()
+    runEvery5Minutes(getHistorical)
+    runEvery3Hours(refreshAuthToken)
+    runEvery3Hours(getKwhr)
+    def rate = settings.samplesPerMinute
+    //log.debug("Sampling at ${rate} samples per minute")
+    schedule("* * * * * ?", doPoll, [data: [cycles: rate]])
```

Comment:
use runEvery1Minute(method, options = [:]) instead of schedule.

Reply:
Is the problem here that we're polling our servers too often, and we shouldn't do it more frequently than every five minutes, or is this something wrong with the schedule() function?

I don't totally understand the comment "have a callback method the server can call to updates the samples."

Reply:
The problem is the load that these frequent scheduling are adding to the platform. Only in a very few cases do we poll anything from the cloud more than every 5 minutes.
By defining a callback/subscription you wouldn't need to have any polls scheduled as the server will call the SA whenever there's a reason to update anything.